datafusion/core/src/physical_plan/mod.rs

```diff
@@ -122,10 +122,20 @@ pub trait ExecutionPlan: Debug + Send + Sync {
     /// have any particular output order here
     fn output_ordering(&self) -> Option<&[PhysicalSortExpr]>;
 
-    /// Specifies the data distribution requirements of all the
-    /// children for this operator
-    fn required_child_distribution(&self) -> Distribution {
-        Distribution::UnspecifiedDistribution
+    /// Specifies the data distribution requirements for all the
+    /// children for this operator, By default it's [[Distribution::UnspecifiedDistribution]] for each child,
+    fn required_input_distribution(&self) -> Vec<Distribution> {
```

Comment:
Maybe we can just use
`vec![Distribution::UnspecifiedDistribution; self.children().len()]`

Reply:
I personally think @yahoNanJing 's suggestion of

```rust
vec![Distribution::UnspecifiedDistribution; self.children().len()]
```

would make the intent clearer

Reply:
Sure, I will change it in the following PR.